### PR TITLE
Relocating Search Bar

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
In this PR we're relocating the Search Bar back into the Notes List.

cc @eshurakov Ready for 👀 sir!! Thanks in advance!!
cc @SylvesterWilmott 

Ref. #626

### Test: Searching
- [x] Verify entering any keyword in the Search Bar yields global results
- [x] Verify the Minimap reflects search results
- [x] Verify the Editor's Keyword Highlight works super
- [x] Verify clicking over the Editor when Search is active causes the Highlights to be dropped
- [x] Verify clearing the Search Bar's contents causes the Notes List to display the documents associated to the current selection

### Test: Dismissal
- [x] Verify clicking over any Tag / Trash causes the Search to be dismissed
- [x] Verify that clicking over the `+` New Note button causes the Search to be dismissed

### Test: Logout
1. Enter any random keyword in the search field
2. Logout
3. Log in again

- [x] Verify the Search Field is cleared out

### Release
`RELEASE-NOTES.txt` was updated in b08d1665dc276da8e2a640440549b10b0d57df21 with:

> SearchBar has been relocated into the Notes List #838

### CODE REVIEW

File: Simplenote/NoteEditorViewController.h [hunk 1]
```diff
@@ -79,9 +74,11 @@ typedef NS_ENUM(NSInteger, NoteFontSize) {
 @property (nonatomic,   weak) Note                                              *note;
 @property (nonatomic,   weak) id<EditorControllerNoteActionsDelegate>           noteActionsDelegate;
 @property (nonatomic,   weak) id<EditorControllerTagActionsDelegate>            tagActionsDelegate;
-@property (nonatomic,   weak) id<EditorControllerSearchDelegate>                searchDelegate;
 @property (nonatomic, strong, nullable) SearchMapView                           *searchMapView;
 
+// TODO: Switch NSObject >> SearchQuery. ObjC compiler isn't picking up the Swift Package =(
+@property (nonatomic, strong, nullable) NSObject                                *searchQuery;
```
Comment (eshurakov) [inline Simplenote/NoteEditorViewController.h:80]: Had the same thing on iOS :-/

**Review by eshurakov — APPROVED** [review]
Nice! Looks good :-)

:shipit:

**Comment by eshurakov** [comment]
Just noticed, that in dark mode the bottom border of search bar is thicker than the top one. Not sure if that's how it should be.

Screenshot 2021-02-04 at 11 50 29

**Comment by SylvesterWilmott** [comment]
Thanks!

**Comment by jleandroperez** [comment]
Thank you Gentleman!!!

@eshurakov We're overriding the SearchBar standard draw elsewhere. It's gonna be a fun thing to do (I hope?)

Merging this one!

🔥

**Comment by pachlava** [comment]
[testing] fix/feature verified on MBP 16" Intel | macOS 11.2 | Simplenote 2.8 (11218)

**Pull request merged**